upstash_redis/typing.py

```diff
@@ -1,10 +1,17 @@
-from typing import List, Literal, TypedDict, Union
+from typing import List, Literal, TypedDict, Union, Dict
 
 """
 The type of the "result" field returned by the REST API. 
 """
 RESTResultT = Union[str, int, List, None, Literal[0, 1], Literal["OK"]]
 
+"""
+Types that are automatically converted into string before getting sent to redis.
+
+So they are still stored as strings. This is just a shorthand for the user.
+"""
+Value = Union[str, int, float, bool, None, List["Value"], Dict[str, "Value"]]
```

Comment:
Can you rename this to `ValueT`, so that it would be similar to others?

Reply:
Also, this implies wrong things for some commands. For example, you should not do redis.set() with value of type List[Value]

Reply:
Yes i agree, should i make it an exception an leave redis.set as `str | int | float | bool`?

Reply:
No, I don't think that would be enough. I give set as an example, but there are lots of similar places. Why do we need to have List and Dict in the Value type? Which commands require it?

Reply:
```py
command = [
  element
  if (
      isinstance(element, str)
      or isinstance(element, int)
      or isinstance(element, float)
  )
  else dumps(element)
  for element in command
]
```

Because of this, i thought i should accept json like data too. But I can remove List and Dict it from the type if its ok.

Reply:
Yeah, I think it would be best to remove List, Dict, and None